installer/engine_installer/server.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"html/template"
+	"net/http"
+	"os"
+	"syscall"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/vic/pkg/certificate"
+	"github.com/vmware/vic/pkg/trace"
+)
+
+const (
+	publicNetName = "public-net"
+	bridgeNetName = "bridge-net"
+	imgStoreName  = "img-store"
+	computeName   = "compute"
+)
+
+var (
+	engineInstaller = NewEngineInstaller()
+	c               config
+)
+
+type config struct {
+	addr     string
+	cert     tls.Certificate
+	serveDir string
+}
+
+func Init(conf *config) {
+	ud := syscall.Getuid()
+	gd := syscall.Getgid()
+	log.Info(fmt.Sprintf("Current UID/GID = %d/%d", ud, gd))
+
+	flag.StringVar(&conf.addr, "addr", ":1337", "Listen address - must include host and port (addr:port)")
+	flag.StringVar(&conf.serveDir, "data", "/opt/vmware/engine_installer", "Directory containing vic-machine and HTML data")
+
+	flag.Parse()
+
+	generateCert(conf)
+}
+
+func main() {
+	Init(&c)
+
+	defer trace.End(trace.Begin(""))
```

Comment:
Not sure if using a trace on `main()` is necessary. It should be moved above the call to `Init` if it's being used.